src/test/java/org/opensearch/sdk/TestExtensionRestPathRegistry.java

```diff
@@ -0,0 +1,99 @@
+
+public class TestExtensionRestPathRegistry extends OpenSearchTestCase {
+
+    private ExtensionRestPathRegistry extensionRestPathRegistry = new ExtensionRestPathRegistry();
+
+    private ExtensionRestHandler fooHandler = new ExtensionRestHandler() {
+        @Override
+        public List<Route> routes() {
+            return List.of(new Route(Method.GET, "/foo"));
+        }
+
+        @Override
+        public ExtensionRestResponse handleRequest(Method method, String uri) {
+            return null;
+        }
+    };
+    private ExtensionRestHandler barHandler = new ExtensionRestHandler() {
+        @Override
+        public List<Route> routes() {
+            return List.of(new Route(Method.PUT, "/bar/{planet}"));
+        }
+
+        @Override
+        public ExtensionRestResponse handleRequest(Method method, String uri) {
+            return null;
+        }
+    };
+    private ExtensionRestHandler bazHandler = new ExtensionRestHandler() {
+        @Override
+        public List<Route> routes() {
+            return List.of(new Route(Method.POST, "/baz/{moon}/qux"), new Route(Method.PUT, "/bar/baz"));
+        }
+
+        @Override
+        public ExtensionRestResponse handleRequest(Method method, String uri) {
+            return null;
+        }
+    };
+
+    @Override
+    @BeforeEach
+    public void setUp() throws Exception {
+        List<ExtensionRestHandler> handlerList = List.of(fooHandler, barHandler, bazHandler);
+        super.setUp();
+        for (ExtensionRestHandler handler : handlerList) {
+            for (Route route : handler.routes()) {
+                extensionRestPathRegistry.registerHandler(route.getMethod(), route.getPath(), handler);
+            }
+        }
+    }
+
+    @Test
+    public void testRegisterConflicts() {
+        // Can't register same exact name
+        assertThrows(IllegalArgumentException.class, () -> extensionRestPathRegistry.registerHandler(Method.GET, "/foo", fooHandler));
+        // Can't register conflicting named wildcards
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> extensionRestPathRegistry.registerHandler(Method.PUT, "/bar/{none}", barHandler)
+        );
+    }
+
+    @Test
+    public void testGetHandler() {
+        assertEquals(fooHandler, extensionRestPathRegistry.getHandler(Method.GET, "/foo"));
+        assertNull(extensionRestPathRegistry.getHandler(Method.PUT, "/foo"));
+
+        // Exact match and wildcard match can overlap, exact takes priority
+        assertEquals(barHandler, extensionRestPathRegistry.getHandler(Method.PUT, "/bar/mars"));
+        assertEquals(bazHandler, extensionRestPathRegistry.getHandler(Method.PUT, "/bar/baz"));
+        assertNull(extensionRestPathRegistry.getHandler(Method.PUT, "/bar/mars/bar"));
+
+        assertEquals(bazHandler, extensionRestPathRegistry.getHandler(Method.POST, "/baz/europa/qux"));
+        assertNull(extensionRestPathRegistry.getHandler(Method.POST, "/bar/europa"));
```

Comment:
I was like why doesn't this work and I realized the Method is POST. 
For few paths, folks usually want to support multiple Methods like GET, DELETE etc. 
Is this something you plan on to add ?

Reply:
The capability is there, you just add it in `routes()` and handle it appropriately in `handleRequest()`.   I'm trying not to overcomplicate hello world, though! :)